src/Buffer.h

```diff
@@ -2,144 +2,107 @@
 #define HALIDE_BUFFER_H
 
 /** \file
- * Defines Buffer - A c++ wrapper around a buffer_t.
+ * Defines BufferPtr - A c++ wrapper around a buffer_t.
```

Comment:
Done, because it might actually make the diff easier to read. BufferPtr is only superficially similar to Buffer